x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -72,18 +73,33 @@ interface IWaterfallItemBase {
   childIds?: Array<IWaterfallItemBase['id']>;
 }
 
-interface IWaterfallItemTransaction extends IWaterfallItemBase {
+export interface IWaterfallItemTransaction extends IWaterfallItemBase {
   transaction: Transaction;
   docType: 'transaction';
   errorCount: number;
 }
 
-interface IWaterfallItemSpan extends IWaterfallItemBase {
+export interface IWaterfallItemSpan extends IWaterfallItemBase {
   span: Span;
   docType: 'span';
 }
 
-export type IWaterfallItem = IWaterfallItemSpan | IWaterfallItemTransaction;
+export interface IWaterfallItemAgentMark extends IWaterfallItemBase {
+  docType: 'agentMark';
+}
+
+export interface IWaterfallItemError extends Omit<IWaterfallItemBase, 'name'> {
```

Comment:
Is it necessary to omit `name`?

Reply:
I don't see you doing that for `IWaterfallItemAgentMark `

Reply:
I did it for Error because it doesn't have any tag to be used as a name, but it does for Agent Mark, which uses the name of the metric as a name.